doc/guides/networking/net-stack-architecture.rst

```diff
@@ -0,0 +1,137 @@
+.. _network_stack_architecture:
+
+Network Stack Architecture
+##########################
+
+High level overview of the network stack
+****************************************
+
+.. figure:: zephyr_netstack_overview.svg
+    :align: left
+    :alt: Overview of the network stack architecture
+    :figclass: align-center
+
+    Network stack overview
+
+*The network stack is layered and it consists of the following parts:*
+
+* **Network Application.** The network application uses
+  :ref:`BSD socket API <bsd_sockets_interface>` to create a network connection,
+  send or receive data, and close a connection. The application can also use
+  the :ref:`network management API <net_mgmt_interface>` to set network
+  related parameters such as taking network interface down, starting a scan
+  (when applicable), setting IP address to a network interface, etc.
+
+* **Network Protocols.** This provides implementations for
+  various protocols such as IPv6, IPv4, UDP, TCP, ICMPv4 and ICMPv6.
+  See :ref:`application protocols chapter <net_protocols>` for information
+  about CoAP, LWM2M and MQTT.
+
+* **Network Interface Abstraction.** This provides functionality
+  that is common in all the network interfaces, such as setting network
+  interface down, etc. There can be multiple network interfaces in the system.
+  See :ref:`network interface overview <net_if_interface>` for more details.
+
+* **L2 Network Technologies.** This provides common API for sending and
+  receiving data to and from an actual network device.
+  See :ref:`L2 overview <net_l2_interface>` for more details.
+  These network technologies include :ref:`Ethernet <ethernet_interface>`,
+  :ref:`IEEE 802.15.4 <ieee802154_interface>`, Bluetooth, CANBUS, etc.
+  Some of these technologies support IPv6 header compression (6Lo), which is
+  done in its own layer. For example ARP for IPv4 is done by the
+  :ref:`Ethernet component <ethernet_interface>`.
+
+* **Network Device Driver.** The actual low-level device driver handles the
```

Comment:
Suggested to use plural here (device driver**s**, etc.), to make it obvious that multiple devices are supported.